src/hotspot/share/classfile/loaderConstraints.cpp

```diff
@@ -185,62 +191,54 @@ class PurgeUnloadedConstraints : public StackObj {
     LogTarget(Info, class, loader, constraints) lt;
     int len = set.num_constraints();
     for (int i = len - 1; i >= 0; i--) {
-      LoaderConstraint* probe = set.constraint_at(i);
-      InstanceKlass* klass = probe->klass();
-      // Remove klass that is no longer alive
-      if (klass != NULL &&
-          !klass->is_loader_alive()) {
-        probe->set_klass(NULL);
+      LoaderConstraint* constraint = set.constraint_at(i);
+      InstanceKlass* klass = constraint->klass();
+
+      // Remove klass and constraint that is no longer alive
+      if (klass != NULL && !klass->is_loader_alive()) {
         if (lt.is_enabled()) {
           ResourceMark rm;
           lt.print("purging class object from constraint for name %s,"
```

Comment:
You are doing more than just purging the class object now.

Reply:
Right.  This purges the entire constraint for the unloaded class object, without looping through the loader entries and doing it separately.  They should already be purged.

Reply:
My point being the log message is no longer accurate.